pkg/applyconfigurations/testdata/ac/zz_generated.applyconfigurations.go

```diff
@@ -0,0 +1,441 @@
+	v1 "k8s.io/client-go/applyconfigurations/core/v1"
+	applyconfigurationsmetav1 "k8s.io/client-go/applyconfigurations/meta/v1"
+	testdata "sigs.k8s.io/controller-tools/pkg/applyconfigurations/testdata"
+)
+
+// AssociativeTypeApplyConfiguration represents a declarative configuration of the AssociativeType type for use
+// with apply.
+type AssociativeTypeApplyConfiguration struct {
+	Name      *string `json:"name,omitempty"`
+	Secondary *int    `json:"secondary,omitempty"`
+	Foo       *string `json:"foo,omitempty"`
+}
+
+// WithName sets the Name field in the declarative configuration to the given value
+func (b *AssociativeTypeApplyConfiguration) WithName(value string) *AssociativeTypeApplyConfiguration {
+	b.Name = &value
+	return b
+}
+
+// WithSecondary sets the Secondary field in the declarative configuration to the given value
+func (b *AssociativeTypeApplyConfiguration) WithSecondary(value int) *AssociativeTypeApplyConfiguration {
+	b.Secondary = &value
+	return b
+}
+
+// WithFoo sets the Foo field in the declarative configuration to the given value
+func (b *AssociativeTypeApplyConfiguration) WithFoo(value string) *AssociativeTypeApplyConfiguration {
+	b.Foo = &value
+	return b
+}
+
+// AssociativeTypeApplyConfiguration represents a declarative configuration of the AssociativeType type for use
+// with apply.
+func AssociativeType() *AssociativeTypeApplyConfiguration {
+	return &AssociativeTypeApplyConfiguration{}
+}
+
+// CronJobApplyConfiguration represents a declarative configuration of the CronJob type for use
+// with apply.
+type CronJobApplyConfiguration struct {
+	applyconfigurationsmetav1.TypeMetaApplyConfiguration    `json:",inline"`
+	*applyconfigurationsmetav1.ObjectMetaApplyConfiguration `json:"metadata,omitempty"`
+	Spec                                                    *CronJobSpecApplyConfiguration   `json:"spec,omitempty"`
+	Status                                                  *CronJobStatusApplyConfiguration `json:"status,omitempty"`
+}
+
+// WithSpec sets the Spec field in the declarative configuration to the given value
+func (b *CronJobApplyConfiguration) WithSpec(value *CronJobSpecApplyConfiguration) *CronJobApplyConfiguration {
+	b.Spec = value
+	return b
+}
+
+// WithStatus sets the Status field in the declarative configuration to the given value
+func (b *CronJobApplyConfiguration) WithStatus(value *CronJobStatusApplyConfiguration) *CronJobApplyConfiguration {
+	b.Status = value
+	return b
+}
+
+// CronJobApplyConfiguration represents a declarative configuration of the CronJob type for use
+// with apply.
+func CronJob() *CronJobApplyConfiguration {
+	return &CronJobApplyConfiguration{}
+}
+func (ac *CronJobApplyConfiguration) GetReferenceType() runtime.Object {
+	return &testdata.CronJob{}
+}
+
+// CronJobListApplyConfiguration represents a declarative configuration of the CronJobList type for use
+// with apply.
+type CronJobListApplyConfiguration struct {
```

Comment:
Can this generated type ever be used? For k/k we only generated types that would be reachable from the root types that could be passed to "Apply".